### ISSUE TITLE
wp.components.SelectControl not working correctly

### ISSUE THREAD

**Issue by eduardoescdel** [WordPress/gutenberg#21965]
I'm building a WP-Admin js script for customizing my block editor, i'm using **editor.BlockEdit** filter to do so.

While creating a new select element with **wp.element.createElement(wp.components.SelectControl, ...);** it seems to displays correctly in the block settings area, however there is a strange behaviour; if i include a default **value** property for the select element, then it won't update after i click on any of the options. If i don't include any **value** property, then the select element bevahes normally, it will show up the selected option i clicked on.

```javascript
wp.element.createElement(
    wp.components.SelectControl,
    {
        value: 1, // Setting up this property produces the described misbehaviour
        label: "Hello World",
        options: [
            {
                label: "One",
                value: 1
            },
            {
                label: "Two",
                value: 2
            }
        ],
        onChange: function( value ) {
            // Do something with current block props
        }
    }
);
```
I've been several hours dealing with this, i even tried the long way, using **wp.element.createElement('select', ...);** to no avail, there just seems to be that the **value** messes everything up.

I'm using Wordpress 5.4 with "Twenty Twenty" theme, everything is up to date. I see no errors in the js console.

**Comment by talldan** [comment]
@eduardoescdel The `value` prop doesn't represent default value, but it should represent what the value currently is, e.g. the code should be:
```js
wp.element.createElement(
    wp.components.SelectControl,
    {
        value: currentValue,
        label: "Hello World",
        options: [
            {
                label: "One",
                value: 1
            },
            {
                label: "Two",
                value: 2
            }
        ],
        onChange: function( value ) {
            // Set the current value
        }
    }
);
```

In terms of blocks, this usually entails passing the attribute value into SelectControl as the value prop.

For further details on the subject, I'd recommend looking into the difference between 'uncontrolled' and 'controlled' components in React:
- https://reactjs.org/docs/uncontrolled-components.html
- https://reactjs.org/docs/forms.html#controlled-components

Basically, when the value is unset the SelectControl is considered uncontrolled, but when it's set it's considered controlled.

**Comment by eduardoescdel** [comment]
@talldan thank you. I'm not really a react guy, sorry. I investigated a little about controlled / uncontrolled components and found a solution; i was missing the **wp.compose.withState()** wrapper (which maybe is what you call "implementor"?).

Anyway, thank you again for enlight me!
